### PR TITLE
Add CORS header to generic internal server error response

### PR DESCRIPTION
**Pull request by matheus23**
This is implementing option 2 from #491

I don't expect this to be merged, this is just me trying to see what this solution looks like.

This gets rid of unexpected CORS errors on the front-end when there's an internal server error. So instead of a CORS error it now looks like this in front-ends:

image

Another benefit is that anything that's creating a bad request will now also return with CORS headers.

I'm pretty sure having an accept-anything CORS header is fine here, as the response will never reveal any information, only a status code + a "Bad Request" or "Something went wrong" body.

### CODE REVIEW

File: fission-web-server/library/Fission/Web/Server/Internal/Production.hs [hunk 1]
```diff
@@ -215,6 +216,10 @@ mkSettings logger (Server.Port port) =
   defaultSettings
     |> setPort port
     |> setLogger (Web.Log.fromLogFunc logger)
+    |> setOnExceptionResponse
+      (mapResponseHeaders (("Access-Control-Allow-Origin", "*") :) 
+        . defaultOnExceptionResponse
+      )
```
Comment (expede) [inline fission-web-server/library/Fission/Web/Server/Internal/Production.hs:222]: I also have a tendency for point-free, but outside of a few cases, adding the lambda makes this stuff much more legible. A named function is even better:

```haskell
-- Why isn't this in the WAI package? Seems useful
addResponseHeader :: HeaderName -> ByteString -> Response -> Response
addResponseHeader field val = mapResponseHeaders \headers -> (field, val) : headers

-- ...

|> setLogger (Web.Log.fromLogFunc logger)
|> setOnExceptionResponse \err -> addResponseHeader $ defaultOnExceptionResponse err
|> setTimeout serverTimeout
```

**Review by expede — APPROVED** [review]
Just the one suggestion, but otherwise LGTM 👍

**Comment by expede** [comment]
Aaaaaaand as usual, please bump the version number before merging

File: fission-web-server/library/Fission/Web/Server/Internal/Production.hs (hunk 1 above)
Comment (matheus23) [reply fission-web-server/library/Fission/Web/Server/Internal/Production.hs:222]: oooooh I always forget that we have BlockArguments! That actually looks super clean!

Comment (matheus23) [reply fission-web-server/library/Fission/Web/Server/Internal/Production.hs]: Ahh, that actually doesn't work :slightly_frowning_face: Apparently, it'll put the next `|>` into the lambda.

**Comment by matheus23** [comment]
Closes #491

**Pull request merged**